library/std/src/net/ip.rs

```diff
@@ -1509,12 +1505,12 @@ impl Ipv6Addr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn to_ipv4(&self) -> Option<Ipv4Addr> {
-        match self.segments() {
-            [0, 0, 0, 0, 0, f, g, h] if f == 0 || f == 0xffff => {
-                Some(Ipv4Addr::new((g >> 8) as u8, g as u8, (h >> 8) as u8, h as u8))
-            }
-            _ => None,
+        if let [0, 0, 0, 0, 0, 0 | 0xffff, ab, cd] = self.segments() {
+            let [a, b] = ab.to_be_bytes();
+            let [c, d] = cd.to_be_bytes();
+            return Some(Ipv4Addr::new(a, b, c, d));
         }
+        None
```

Comment:
```suggestion
            Some(Ipv4Addr::new(a, b, c, d))
        } else {
            None
        }
```

Reply:
thank you. I think the current form is nicer as it doesn't require else block.

Reply:
I don't feel strongly about this, but I do think it's more idiomatic to implicitly return an `if`-`else` expression.